src/repos/WeeklyDebriefRepo.ts

```diff
@@ -0,0 +1,55 @@
+import { DocumentType } from '@typegoose/typegoose';
+import { User, UserModel } from '../entities/User';
+import WeeklyDebrief from '../entities/WeeklyDebrief';
+import { ArticleModel } from '../entities/Article';
+
+const getExpirationDate = (creationDate: Date): Date => {
+  creationDate.setDate(creationDate.getDate() + 7);
+  return creationDate;
+};
+
+const createWeeklyDebrief = async (
+  user: DocumentType<User>,
+  creationDate: Date,
+  expirationDate: Date,
+): Promise<User> => {
+  const articleAggregate = ArticleModel.aggregate();
+  const { uuid } = user;
+  const weeklyDebrief = Object.assign(new WeeklyDebrief(), {
+    uuid,
+    creationDate,
+    expirationDate,
+    numShoutouts: user.numShoutouts,
+    numBookmarkedArticles: user.numBookmarkedArticles,
+    readArticles: user.readArticles.slice(0, 2),
+    numReadArticles: user.readArticles.length,
```

Comment:
Why are only the first two articles of readArticles stored in readArticles? It seems a little strange to have a readArticles and numReadArticles - why store 2 fields when just storing `user.readArticles` and performing the `.slice` and `.length` operator later would keep the model simpler and make us store less information.

Reply:
On the [weekly debrief UI](https://www.figma.com/file/vMfqUmHwm9VHGChOjPQFlf/?node-id=3%3A2), we only show two of the articles that a user read over the course of the past week for sharing or revisiting. The `numReadArticles` field shows the user the summarized statistic of _how many_ articles they read over the last week.

Reply:
That's not my point, storing `readArticles` and `numReadArticles` is redundant, we just want to store `user.readArticles` since if we want to get what we're currently storing under `readArticles` and `numReadArticles` when can then just do `.slice(0,2)` and `.length`. If this is more intuitive to you @kidzegeye then feel free to keep it but I'd definitely be confused seeing that the user's `readArticles` and a weekly debrief's `readArticles` are not the same thing especially when this behavior is undocumented. If they were the same and then when I get a weekly debrief I do the splicing and the length operation and have it commented in the function doing it, then it is much clearer. Think about what we'll need to do in the future if we decide to have the weekly debrief UI show the three of the articles a user read over the course of the past week rather than two - now you're gonna have to manually update each of the objects and that will be a nightmare. This should be fixed before I approve this, this will lead to inflexibility in the future.